oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/FormattingUtils.java

```diff
@@ -18,18 +18,28 @@
 
 import org.apache.jackrabbit.guava.common.base.Stopwatch;
 
-import java.time.LocalTime;
-import java.time.format.DateTimeFormatter;
 import java.util.concurrent.TimeUnit;
 
 public class FormattingUtils {
     public static String formatToSeconds(Stopwatch stopwatch) {
-        LocalTime seconds = LocalTime.ofSecondOfDay(stopwatch.elapsed(TimeUnit.SECONDS));
-        return DateTimeFormatter.ISO_TIME.format(seconds);
+        long seconds = stopwatch.elapsed(TimeUnit.SECONDS);
+        long absSeconds = Math.abs(seconds);
+        return String.format(
+                "%02d:%02d:%02d",
+                absSeconds / 3600,
```

Comment:
The formulas look correct, but it should be possible to use Java utilities...

```
long millis = -123456789;
long hours = Math.abs(TimeUnit.MILLISECONDS.toHours(millis));
long minutes = Math.abs(TimeUnit.MILLISECONDS.toMinutes(millis)) % 60;
long seconds = Math.abs(TimeUnit.MILLISECONDS.toSeconds(millis)) % 60;
long milliseconds = Math.abs(millis) % 1000;

String formattedTime = String.format("%s%02d:%02d:%02d.%03d", millis < 0 ? "-" : "", hours, minutes, seconds, milliseconds);
System.out.println(formattedTime);
```

Reply:
Yes, it becomes easier to read. I have refactored the code.